src/mock/ray/observability/BUILD.bazel

```diff
@@ -0,0 +1,9 @@
+load("//bazel:ray.bzl", "ray_cc_library")
+
+ray_cc_library(
+    name = "mock_metric_interface",
+    hdrs = ["metric_interface.h"],
+    deps = [
+        "//src/ray/observability:ray_event_recorder_interface",
+    ],
+)
```

Comment:
same comment as the other PR, let's make this a "fake" implementation (not using gtest) and move it into the source tree (so we'll have `metric_interface.h`, `metric.h`, and `fake_metric.h`)